docs/python_support/python_support.md

```diff
@@ -0,0 +1,333 @@
+
+ Starting with version 2023.3, model server introduced support for execution of custom Python code. Users can now create their own servables in Python. Those servables can run simple pre or post processing tasks as well as complex ones like image or text generation. 
+ 
+ Python is enabled via [Mediapipe](../mediapipe.md) by built-in `PythonExecutorCalculator` that allows creating graph nodes that execute Python code. That way Python servables can be used as standalone endpoints (single node graphs) or be part of larger Mediapipe solutions.
+
+ Checkout [quickstart guide](quickstart.md) for simple usage example.
+
+*Note:* This feature is currently in preview.
+
+ ## Building Docker Image
+
+Publically distributed Docker images support Python, but they do not come with any external modules. If bare Python is all you need then you can use public image directly. Otherwise you need to extend public image with additional layers that will install all the stuff you need for your Python code to work. For example, let's say you need numpy.
+In that case your Dockerfile may look like:
+
+```dockerfile
+FROM openvino/model_server:latest
+USER root
+ENV LD_LIBRARY_PATH=/ovms/lib
+ENV PYTHONPATH=/ovms/lib/python
+RUN apt update && apt install -y python3-pip git
+RUN pip3 install numpy
+ENTRYPOINT [ `/ovms/bin/ovms` ]
+```
+
+TODO: Here we could provide make target with requirements location as a parameter.
+
+### Building Model Server From Source
+In above section you use public Docker image. In case you want to build it from source do the following:
+
+```bash
+git clone https://github.com/openvinotoolkit/model_server.git
+cd model_server
+make docker_build MEDIAPIPE_DISABLE=0 PYTHON_DISABLE=0 OV_USE_BINARY=1 RUN_TESTS=0
+cd ..
+```
+Resulting Docker image can be extended with additional layers just as the public one.
+
+## Python Servable
+
+When deploying Python servable, Model Server expects Python file with an `OvmsPythonModel` class implemented:
+
+```python
+class OvmsPythonModel:
+
+    def initialize(self, kwargs):
+        """
+        `initialize` is called when model server loads graph definition. It allows to initialize and maintain state between subsequent execute() calls and even graph instances. 
+        For gRPC unary, graphs are recreated per request. 
+        For gRPC streaming, there can be multiple graph instances existing at the same time. 
+        OvmsPythonModel object is initialized with this method and then shared between all graph instances. Implementing this function is required (should it be?).
+
+        Parameters:
+        -----------
+        kwargs : dict
+            Available arguments:
+            * node_name: string
+                Name of the node in the graph
+            * input_streams: list of strings
+                List of input stream names defined for the node in graph configuration
```

Comment:
I think this is wrong to say "input stream names", maybe we should explain the A:B format user received? rhs/lhs

Reply:
ideally show part of pbtxt file here and what python script will receive

Reply:
Let's do that in the descriptions section below. I don't want to make this Python code sample too big.